fiat-go/64/p521/p521.go

```diff
@@ -1,45 +1,43 @@
-
-   balance = [0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x3fffffffffffffe]
- */
+// Autogenerated: 'src/ExtractionOCaml/unsaturated_solinas' --lang Go --no-wide-int --cmovznz-by-mul --internal-static --package-case flatcase --public-function-case UpperCamelCase --private-function-case camelCase --public-type-case UpperCamelCase --private-type-case camelCase --no-prefix-fiat --package-name p521 '' 64 9 '2^521 - 1' carry_mul carry_square carry add sub opp selectznz to_bytes from_bytes
+//
+// curve description (via package name): p521
+//
+// machine_wordsize = 64 (from "64")
+//
+// requested operations: carry_mul, carry_square, carry, add, sub, opp, selectznz, to_bytes, from_bytes
+//
+// n = 9 (from "9")
+//
+// s-c = 2^521 - [(1, 1)] (from "2^521 - 1")
+//
+// tight_bounds_multiplier = 1 (from "")
+//
+//
+//
+// Computed values:
+//
+// carry_chain = [0, 1, 2, 3, 4, 5, 6, 7, 8, 0, 1]
+//
+// eval z = z[0] + (z[1] << 58) + (z[2] << 116) + (z[3] << 174) + (z[4] << 232) + (z[5] << 0x122) + (z[6] << 0x15c) + (z[7] << 0x196) + (z[8] << 0x1d0)
+//
+// bytes_eval z = z[0] + (z[1] << 8) + (z[2] << 16) + (z[3] << 24) + (z[4] << 32) + (z[5] << 40) + (z[6] << 48) + (z[7] << 56) + (z[8] << 64) + (z[9] << 72) + (z[10] << 80) + (z[11] << 88) + (z[12] << 96) + (z[13] << 104) + (z[14] << 112) + (z[15] << 120) + (z[16] << 128) + (z[17] << 136) + (z[18] << 144) + (z[19] << 152) + (z[20] << 160) + (z[21] << 168) + (z[22] << 176) + (z[23] << 184) + (z[24] << 192) + (z[25] << 200) + (z[26] << 208) + (z[27] << 216) + (z[28] << 224) + (z[29] << 232) + (z[30] << 240) + (z[31] << 248) + (z[32] << 256) + (z[33] << 0x108) + (z[34] << 0x110) + (z[35] << 0x118) + (z[36] << 0x120) + (z[37] << 0x128) + (z[38] << 0x130) + (z[39] << 0x138) + (z[40] << 0x140) + (z[41] << 0x148) + (z[42] << 0x150) + (z[43] << 0x158) + (z[44] << 0x160) + (z[45] << 0x168) + (z[46] << 0x170) + (z[47] << 0x178) + (z[48] << 0x180) + (z[49] << 0x188) + (z[50] << 0x190) + (z[51] << 0x198) + (z[52] << 0x1a0) + (z[53] << 0x1a8) + (z[54] << 0x1b0) + (z[55] << 0x1b8) + (z[56] << 0x1c0) + (z[57] << 0x1c8) + (z[58] << 0x1d0) + (z[59] << 0x1d8) + (z[60] << 0x1e0) + (z[61] << 0x1e8) + (z[62] << 0x1f0) + (z[63] << 0x1f8) + (z[64] << 2^9) + (z[65] << 0x208)
+//
+// balance = [0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x7fffffffffffffe, 0x3fffffffffffffe]
 package p521
```

Comment:
Oh, also, it would be best to detach the comment from the `package` line, because if one tries to put multiple fields in a single package (using the curve prefix, which is what I am leaning towards at the moment) the comments will conflict. If you want it to show up in godocs, maybe you can attach it to a type.

```suggestion
// ...

package p521
```

Reply:
>  if one tries to put multiple fields in a single package (using the curve prefix, which is what I am leaning towards at the moment) the comments will conflict

Can you explain this?

Reply:
If I put `p521.go` and `p384.go` in the same package, both generated with fiat (using the same package name and a non-empty curve description), everything works except that there are two comments for `package fiat`, which is a conflict. If you add a newline before the `package` line, it stops being a package-global godoc comment, and two implementations can coexist in the same package.

Reply:
But they have different package names?  Are you talking about the 32 vs 64 bit versions?  Would it help to pick different package names?  Or is it really best to detach the comment from the doc? (But then it won't show up on pkg.go.dev, right?)

Reply:
What I'm currently doing is I'm generating the fiat code as completely private with this command. (I got the Dockerfile to work reliably, so I'm ok with doing my own generation, but I would like not to change the output for verifiability.)

```
unsaturated_solinas --lang Go --no-wide-int --cmovznz-by-mul \
        --internal-static --public-function-case camelCase --public-type-case camelCase \
        --private-function-case camelCase --private-type-case camelCase \
        --no-prefix-fiat --package-name fiat p521 64 9 '2^521 - 1' \
        carry_mul carry_square carry add sub to_bytes from_bytes
```

Then I'm building wrapper `fiat.P521Element` types around it. Having Fiat set the package docs would be jarring because none of the Fiat functions are exposed this way.

Moreover, this way I can also put `fiat.P521Element` and `fiat.P384Element` in the same package.

Reply:
How do you feel about adding the option `--doc-newline-before-package-declaration` to have this, and letting the code we check in to fiat-crypto itself have attached package docs?